mlir/include/mlir/Dialect/MIOpen/MIOpenOps.td

```diff
@@ -16,6 +16,12 @@
 include "mlir/IR/OpBase.td"
 include "mlir/Interfaces/SideEffectInterfaces.td"
 
+// TODO: upstream this
+class Fixed1DVector<int length, Type type> :
```

Comment:
@krzysz00 could you explain the merit adding this class?

Reply:
The reason this has been added is to allow us to not have to write the type of the vector when we know it's length and size - hence the `SameBuildabilityAs`. We used this inferred vector type in a few places in our formats.